diff --git a/src/gl/fpe_shader.c b/src/gl/fpe_shader.c
index 23068c0..52e074b 100644
--- a/src/gl/fpe_shader.c
+++ b/src/gl/fpe_shader.c
@@ -910,7 +910,7 @@
         int t = state->texture[i].textype;
         if(point && !pointsprite) t=0;
         if(!is_default)
-            if(t && !need->need_texs&(1<<i))
+            if(t && !(need->need_texs&(1<<i)))
                 t = 0;
         if(t) {
             sprintf(buff, "varying highp %s _gl4es_TexCoord_%d;\n", texvecsize[t-1], i);
